src/core/layout/qgscompositionconverter.h

```diff
@@ -0,0 +1,212 @@
+/***************************************************************************
+  qgscompositionconverter.h - Convert  a QGIS 2.x composition to a layout
+
+ ---------------------
+ begin                : 13.12.2017
+ copyright            : (C) 2017 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSCOMPOSITIONCONVERTER_H
+#define QGSCOMPOSITIONCONVERTER_H
+
+
+#include <QDomDocument>
+#include <QDomElement>
+
+#include "qgis.h"
+#include "qgis_sip.h"
+
+#define SIP_NO_FILE
+
+#include "qgspropertycollection.h"
+
+class QgsLayout;
+class QgsLayoutItem;
+class QgsReadWriteContext;
+class QgsProperty;
+class QgsPropertyCollection;
+
+class QgsLayoutItemLabel;
+class QgsLayoutItemShape;
+class QgsLayoutItemPicture;
+class QgsLayoutItemPolygon;
+class QgsLayoutItemPolyline;
+class QgsLayoutItemMap;
+class QgsLayoutItemScaleBar;
+class QgsLayoutItemLegend;
+
+
+/**
+ * QgsCompositionConverter class converts a QGIS 2.x composition to a QGIS 3.x layout
+ * \since QGIS 3.0
+ * \note Not available in Python bindings.
+ * \ingroup core
+ */
+class CORE_EXPORT QgsCompositionConverter
+{
+  public:
+
+    /**
+     * Composition data defined properties for different item types
+     */
+    enum DataDefinedProperty
```

Comment:
Good idea putting this here - it means we could remove some of the deprecated ones from QgsLayoutObject and clean it up